client/asset/dcr/dcr.go

```diff
@@ -2538,8 +2538,10 @@ func (dcr *ExchangeWallet) ValidateSecret(secret, secretHash []byte) bool {
 // the specified swap. The contract and matchTime are provided so that wallets
 // may search for the coin using light filters.
 //
-// If the swap was not funded by this wallet, and it is already spent, this
-// method may return asset.CoinNotFoundError. Compare dcr.externalTxOut.
+// For a non-SPV wallet, if the swap appears spent but it cannot be located in a
+// block with a cfilters scan, this will return asset.CoinNotFoundError. For SPV
+// wallets, it is not an error if the transaction cannot be located SPV wallets
+// cannot see non-wallet transactions until they are mined.
```

Comment:
Right that's true that a full node's mempool could be missing it too.  The idea is that would be more cause for alarm than with and SPV wallet.

The alternative to this conditional return of `asset.CoinNotFoundError` on spv status is to adjust in `core` like you said.  Here:

https://github.com/decred/dcrdex/blob/622e702883f5fcefb96fd2410afd736db3b7d931/client/core/trade.go#L793-L808

That's the error log we get, and it does not record the 0 confs.  Fortunately, this function, `counterPartyConfirms`, does not return an error or anything else that will break anything.  If we change `counterPartyConfirms` (and any other callers of `SwapConfirmations`), we'd want to also change BTC so it starts returning `asset.CoinNotFoundError` instead of `nil`

Reply:
Or we can adjust `SwapConfirmations` implementations to not return any error for coin not found scenarios unless the asset backend is sure the coin really does not exist. Could return -1 confs to signal that the coin is not yet found and may be subsequently found in a retry (spv or not). Then asset backends can additionally log a warning if the coin was not found but it really should have been (e.g. in non-spv modes).